diff --git a/terraform/k8s/archs/arch1/app.install.sh b/terraform/k8s/archs/arch1/app.install.sh
index 15b1afa..fe95982 100755
--- a/terraform/k8s/archs/arch1/app.install.sh
+++ b/terraform/k8s/archs/arch1/app.install.sh
@@ -8,7 +8,16 @@ kubectl config set-context --current --namespace=cloudacademy
 
 # ===========================
 
-echo -e "\nSTEP2: create Ingress resource...\n"
+echo -e "\nSTEP2: create Database resources...\n"
+
+cp ./templates/00_db-setup.yaml ./templates/01_db.yaml ./manifests/
+
+kubectl apply -f ./manifests/00_db-setup.yaml
+kubectl apply -f ./manifests/01_db.yaml
+
+# ===========================
+
+echo -e "\nSTEP3: create Ingress resource...\n"
 
 cp ./templates/1_ingress.yaml ./manifests/
 
@@ -16,7 +25,7 @@ kubectl apply -f ./manifests/1_ingress.yaml
 
 # ===========================
 
-echo -e "\nSTEP3: patch Ingress resource...\n"
+echo -e "\nSTEP4: patch Ingress resource...\n"
 
 # shortcut to reuse the FQDN assigned to the nginx ingress controller ELB
 # not recommended for production environments (proper DNS management should be used - R53 etc)
@@ -29,7 +38,7 @@ kubectl patch ingress public --type json -p "[{\"op\": \"replace\", \"path\": \"
 
 # ===========================
 
-echo -e "\nSTEP4: create API resources...\n"
+echo -e "\nSTEP5: create API resources...\n"
 
 sed \
 -e "s/RDS_AURORA_ENDPOINT/$1/g" \
@@ -39,7 +48,7 @@ kubectl apply -f ./manifests/2_api.yaml
 
 # ===========================
 
-echo -e "\nSTEP5: create APP (frontend) compute resources...\n"
+echo -e "\nSTEP6: create APP (frontend) compute resources...\n"
 
 sed \
 -e "s/INGRESS_FQDN/${INGRESS_LB_FQDN}/g" \
diff --git a/terraform/main.tf b/terraform/main.tf
index d233c0d..2aa314d 100644
--- a/terraform/main.tf
+++ b/terraform/main.tf
@@ -50,23 +50,6 @@ locals {
     max_size       = 2
     desired_size   = 2
   }
-
-  rds = {
-    master_username = "root"
-    master_password = "followthewhiterabbit"
-    db_name         = "cloudacademy"
-    scaling_min     = 2
-    scaling_max     = 4
-  }
-}
-
-#====================================
-
-module "secretsmanager" {
-  source          = "./modules/secretsmanager"
-  master_username = local.rds.master_username
-  master_password = local.rds.master_password
-  db_name         = local.rds.db_name
 }
 
 #====================================
@@ -119,19 +102,6 @@ module "vpc" {
 
 #====================================
 
-module "aurora" {
-  source              = "./modules/aurora"
-  vpc_id              = module.vpc.vpc_id
-  subnet_ids          = module.vpc.private_subnets
-  ingress_cidr_blocks = module.vpc.private_subnets_cidr_blocks
-  master_username     = local.rds.master_username
-  master_password     = local.rds.master_password
-  db_name             = local.rds.db_name
-  secret_manager_arn  = module.secretsmanager.arn
-}
-
-#====================================
-
 module "eks" {
   # forked from terraform-aws-modules/eks/aws, fixes deprecated resolve_conflicts issue
   source = "github.com/cloudacademy/terraform-aws-eks"
@@ -139,9 +109,10 @@ module "eks" {
   cluster_name    = local.k8s.cluster_name
   cluster_version = local.k8s.version
 
+  iam_role_arn = "arn:aws:iam::638744092352:role/cloudacademylabs-EKSClusterRole-RT8EBM6OEM"
   cluster_endpoint_public_access   = true
   attach_cluster_encryption_policy = false
-  create_iam_role                  = true
+  create_iam_role = false
 
   cluster_addons = {
     coredns            = {}
@@ -156,8 +127,9 @@ module "eks" {
 
   eks_managed_node_groups = {
     default = {
+      iam_role_arn = "arn:aws:iam::638744092352:role/cloudacademylabs-EKSClusterNodeGroupRole-MAJNFAKXR1QD"
       use_custom_launch_template = false
-      create_iam_role            = true
+      create_iam_role = false
 
       instance_types = local.k8s.instance_types
       capacity_type  = local.k8s.capacity_type
@@ -232,7 +204,7 @@ resource "terraform_data" "deploy_app" {
       echo "deploying *** ${local.k8s.stocks_app_architecture} *** pattern..."
       rm -f ./manifests/*.yaml
       tree
-      ./app.install.sh ${module.aurora.db_endpoint}
+      ./app.install.sh mysql
     EOT
   }
 
@@ -242,7 +214,6 @@ resource "terraform_data" "deploy_app" {
   }
 
   depends_on = [
-    module.aurora,
     module.eks,
     helm_release.nginx_ingress
   ]
diff --git a/terraform/k8s/archs/arch1/templates/2_api.yaml b/terraform/k8s/archs/arch1/templates/2_api.yaml
index 97b871d..0bf1806 100644
--- a/terraform/k8s/archs/arch1/templates/2_api.yaml
+++ b/terraform/k8s/archs/arch1/templates/2_api.yaml
@@ -1,12 +1,3 @@
-apiVersion: v1
-kind: Secret
-metadata:
-  name: db-creds
-  namespace: cloudacademy
-type: Opaque
-stringData:
-  DB_USER: root
-  DB_PASSWORD: followthewhiterabbit
 ---
 apiVersion: apps/v1
 kind: Deployment
@@ -37,7 +28,7 @@ spec:
         imagePullPolicy: Always
         env:
           - name: DB_CONNSTR
-            value: jdbc:mysql://RDS_AURORA_ENDPOINT:3306/cloudacademy
+            value: jdbc:mysql://mysql:3306/cloudacademy
           - name: DB_USER
             valueFrom:
               secretKeyRef:
